src/Data/Argonaut/Instances/Datetime.purs

```diff
@@ -0,0 +1,51 @@
+module Data.Argonaut.Instances.Datetime (
+  jsDate2Json,
+  json2JSDate
+  ) where
+       
+
+import Data.Function
+import Data.Maybe
+import Data.Either
+
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Decode
+import Data.Date (fromJSDate, Date(), JSDate(), toJSDate)
+
+foreign import jsDate2Json """
+function jsDate2Json(date) {return date;}
+""" :: JSDate -> Json
+
+instance encodeJsonDate :: EncodeJson JSDate where
+  encodeJson = jsDate2Json
+
+instance encodeDate :: EncodeJson Date where
+  encodeJson date = encodeJson $ toJSDate date
+
+foreign import json2JSDateImpl """
+function json2JSDateImpl(json, just, nothing) {
+  if (json && json.constructor && json.constructor == Date) {
+    return just(json); 
+  } else {
+    return nothing;
+  }
+}
+""" :: forall a. Fn3 Json (a -> Maybe a) (Maybe a) (Maybe JSDate)
+
+json2JSDate :: Json -> Maybe JSDate
+json2JSDate json = runFn3 json2JSDateImpl json (Just) (Nothing)
+
+instance decodeJsonJSDate :: DecodeJson JSDate where
```

Comment:
These are currently orphan instances. They should be defined in the `EncodeJson` and `DecodeJson` modules to make sure they are not orphans.